tools/mo/openvino/tools/mo/main.py

```diff
@@ -308,25 +322,53 @@ def raise_ie_not_found():
         from openvino.tools.mo.front.onnx.register_custom_ops import get_front_classes
         import_extensions.load_dirs(argv.framework, extensions, get_front_classes)
 
-    return argv
 
+def check_fallback(argv : argparse.Namespace):
+    fallback_reasons = {}
 
-def prepare_ir(argv):
-    argv = arguments_post_parsing(argv)
+    if not any(deduce_framework_by_namespace(argv)): # no legacy path
+        return fallback_reasons
+
+    if argv.use_new_frontend or argv.use_legacy_frontend: # frontend chosen explicitly
+        return fallback_reasons
+
+    fallback_reasons['extensions'] = \
+        lambda argv : hasattr(argv, 'extensions') and argv.extensions is not None and len(argv.extensions) > 0 \
+            and argv.extensions != import_extensions.default_path() # extensions arg has default value
+    fallback_reasons['transformations_config'] = \
+        lambda argv: hasattr(argv, 'transformations_config') and argv.transformations_config is not None and len(argv.transformations_config) > 0
+
+    reasons = [reason for reason, is_applicable in fallback_reasons.items() if is_applicable(argv)]
+    return reasons
 
+
+def prepare_ir(argv : argparse.Namespace):
+    argv = arguments_post_parsing(argv)
     t = tm.Telemetry()
     graph = None
     ngraph_function = None
     moc_front_end, available_moc_front_ends = get_moc_frontends(argv)
-
     if moc_front_end:
-        t.send_event("mo", "conversion_method", moc_front_end.get_name() + "_frontend")
-        moc_front_end.add_extension(TelemetryExtension("mo", t.send_event, t.send_error, t.send_stack_trace))
-        moc_front_end.add_extension(ProgressReporterExtension(progress_printer(argv)))
-        ngraph_function = moc_pipeline(argv, moc_front_end)
-    else:
-        t.send_event("mo", "conversion_method", "mo_legacy")
-        graph = unified_pipeline(argv)
+        fallback_reasons = check_fallback(argv)
+        if len(fallback_reasons) == 0:
+            t.send_event("mo", "conversion_method", moc_front_end.get_name() + "_frontend")
+            moc_front_end.add_extension(TelemetryExtension("mo", t.send_event, t.send_error, t.send_stack_trace))
+            moc_front_end.add_extension(ProgressReporterExtension(progress_printer(argv)))
+            ngraph_function = moc_pipeline(argv, moc_front_end)
+            return graph, ngraph_function
+        else: # apply fallback
+            reasons_message = ", ".join(fallback_reasons)
+            argv.use_legacy_frontend = True
```

Comment:
I think we don't need to change argv parameters because we throw warning in this case and notify an user about legacy path

Reply:
it is required by `load_extensions` method

Reply:
finally removed, `load_extensions` parameterized in other way